pytab/hwi.py

```diff
@@ -147,19 +191,35 @@ def get_gpu_info() -> list:
 
 def get_cpu_info() -> list:
     cpu_info = cpuinfo.get_cpu_info()
+    # print(cpu_info) # debug
     cpu_elements = list()
-    vendor = cpu_info["vendor_id_raw"]
-    if "intel" in vendor.lower():
-        vendor = "Intel"
-    elif "amd" in vendor.lower() or "advanced micro devices" in vendor.lower():
-        vendor = "Amd"
+
+    # This field might not exist on macOS
+    if "vendor_id_raw" in cpu_info:
+        vendor = cpu_info["vendor_id_raw"]
+        if "intel" in vendor.lower():
+            vendor = "Intel"
+        elif "amd" in vendor.lower() or "advanced micro devices" in vendor.lower():
+            vendor = "Amd"
+
+    elif "Apple" in cpu_info["brand_raw"]:
+        vendor = "Apple"
+
+    else:
+        vendor = "Generic CPU"
```

Comment:
I put in a catch all phrase here in case the script cannot determine the CPU type, so that the script doesn't just fail. There is no reason to not allow CPUs of other brands to run the software tests.

Reply:
Sorry, my bad.
I forgot CPU tests just run.

For some reason I was thinking we do the same Vendor-Type Thing, as we do on GPUs.
There is nothing Stopping us from supporting a "Generic Cpu", you are correct

However it might be useless to Run for a "generic" CPU, since we want to benchmark hardware.
But yes, you are correct!